synfig-studio/src/gui/canvasview.cpp

```diff
@@ -1431,12 +1431,9 @@ CanvasView::on_current_time_widget_changed()
 void
 CanvasView::on_set_end_time_widget_changed()
 {
-	get_canvas()->rend_desc().set_time_end(framedial->get_end_time());
-	//refresh the renddesc
-	refresh_rend_desc();
-	refresh_time_window();
-	//refresh canvas_properties
-	canvas_properties.refresh();
+	RendDesc rend_desc = get_canvas()->rend_desc();
+	rend_desc.set_time_end(framedial->get_end_time());
+	canvas_interface()->set_rend_desc(rend_desc);
```

Comment:
@rodolforg @FirasH 
do we need to create a copy of `rend_desc` here?
I mean, maybe it's more correct to set `time_end` to canvas as well?

```
    RendDesc **&** rend_desc = get_canvas()->rend_desc();
    rend_desc.set_time_end(framedial->get_end_time());
    canvas_interface()->set_rend_desc(rend_desc);
```

Reply:
I don't know if it is good to change it twice:
```c++
    RendDesc **&** rend_desc = get_canvas()->rend_desc();
    rend_desc.set_time_end(framedial->get_end_time()); // HERE
    canvas_interface()->set_rend_desc(rend_desc); // AND HERE
```
What if we improve `canvas_interface()->set_rend_desc()` to not react to an argument equals to current value? Something like:
```c++
if (new_rend_desc == rend_desc)
   return;
. . .
```